ipfs-cluster-service/main.go

```diff
@@ -231,6 +242,74 @@ configuration.
 			Usage:  "run the IPFS Cluster peer (default)",
 			Action: daemon,
 		},
+		{
+			Name:   "migration",
+			Usage:  "migrate the IPFS Cluster state between versions",
+			Description: `
+
+This command migrates the internal state of the ipfs-cluster node to the state 
+specified in a backup file. The state format is migrated from the version 
+of the backup file to the version supported by the current cluster version. 
+To succesfully run a migration of an entire cluster, shut down each peer without
+removal, migrate using this command, and restart every peer.
+`,
+			ArgsUsage: "<BACKUP-FILE-PATH> [RAFT-DATA-DIR]",
+			Action: func(c *cli.Context) error {
+				if c.NArg() < 1 || c.NArg() > 2 {
+					return fmt.Errorf("Usage: <BACKUP-FILE-PATH> [RAFT-DATA-DIR]")
+					
+				}
+				//Load configs
+	                        cfg, clusterCfg, _, _, consensusCfg, _, _, _ := makeConfigs()
+	                        err := cfg.LoadJSONFromFile(configPath)
+ 	                        checkErr("loading configuration", err)
+				backupFilePath := c.Args().First()
+				var raftDataPath string
+				if c.NArg() == 1 {
+					raftDataPath = consensusCfg.DataFolder
+				} else {
+					raftDataPath = c.Args().Get(1)
+				}
+				
+				//Migrate backup to new state
```

Comment:
If you think there is a better file (maybe I should make one?) to execute these things I am open to suggestions.

Reply:
Perhaps let's put most of this action in a `migrate()` function. `migrate.go` file won't hurt either.